src/ripple/app/rdb/impl/Wallet.cpp

```diff
@@ -279,7 +279,11 @@ voteAmendment(
     AmendmentVote vote)
 {
     soci::transaction tr(session);
-    std::string sql =
+    std::string sql = "DELETE FROM FeatureVotes WHERE AmendmentHash='";
```

Comment:
I stumbled upon this while looking for something else. 😆 

I am just wondering: 
- Presumption: `AmendmentHash` is unique for each amendment
- `FeatureVotes` table does not have the `AmendmentHash` as a primary key
- If the `AmendmentHash` was the primary key, `INSERT OR IGNORE INTO` could have been used or `ON CONFLICT` for an upsert.

Reply:
It looks like the query pattern on this table is reading in full and an index would be unnecessary. Just to append @alloynetworks suggestion, the `readAmendments` would have to be modified to incorporate the `RANK()` window function.
```
-- something like this should work
--
SELECT 
  * 
FROM (
   AmendmentHash, 
   AmendmentName, 
   Veto,
   RANK() OVER (
       PARTITION BY AmendmentHash
       ORDER BY ROWID DESC
   ) as rnk FROM FeatureVotes
)
WHERE
    rnk = 1
```

The only consideration would be the growth rate of the table. I presume 30 amendments per year with 80 vote amendments. 2400 per year seems reasonable. 🤔